src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -22,10 +22,12 @@ public class InMemoryExporter<T> : BaseExporter<T>
         where T : class
     {
         private readonly ICollection<T> exportedItems;
+        private readonly bool isMetric;
 
         public InMemoryExporter(ICollection<T> exportedItems)
         {
             this.exportedItems = exportedItems;
+            this.isMetric = typeof(T) == typeof(Metrics.Metric);
```

Comment:
We could refactor `InMemoryExporter` to avoid type-specific checks in a generic class. This could be a follow-up PR.

Reply:
@utpilla what was your idea here?

Reply:
You can use `ConsoleExporter` implementation for reference. You could make `InMemoryExporter<T>` an `abstract` class with the `Export` method as how it is in the `main` branch and have something like this:

```
    public class InMemoryMetricExporter : InMemoryExporter<Metric>
    {
        public InMemoryMetricExporter(ICollection<Metric> exportedItems)
            : base(exportedItems)
        {
        }

        public override ExportResult Export(in Batch<Metric> batch)
        {
            this.exportedItems.Clear();
            foreach (var data in batch)
            {
                this.exportedItems.Add(data);
            }

            return ExportResult.Success;
        }
    }
```

Reply:
@utpilla are we allowed to change the class to abstract, wouldn't that break the public api?
For my learning, is there a process that needs to be followed for such a change?

Reply:
Ahh yes! We could do the same thing without making the class `abstract`. The only thing with that is it would work correctly with the extension method but if a user instantiates a new instance of `InMemoryExporter<Metric>` class they would get the behavior of `InMemoryExporter<T>` and not our custom logic of `InMemoryMetricExporter` class.

Reply:
Hey @utpilla, please review my latest commit.
I made minor changes to the `InMemoryExporter<T>` which allows me to inherit and reuse in `InMemoryMetricExporter`.